rfcs/2022-11-2-14714-expand-vrl-web-ui-beyond-MVP-status.md

```diff
@@ -0,0 +1,147 @@
+# RFC 14714 - 2022-11-02 - Expand VRL web UI beyond MVP status
+
+[The current MVP](https://playground.vrl.dev/) for the VRL web
+playground utilizes WASM and runs the majority of the VRL stdlib,
+[except for six functions](https://github.com/vectordotdev/vector/issues?q=wasm+compatible+label%3A%22vrl%3A+playground%22+is%3Aopen).
+Users can run VRL programs, share their programs online, and try
+sample event inputs. However, we would like to expand the MVP
+features which would require more planning and decision making.
+An outline of what we would like to implement can be found in the
+Context section below.
+
+## Context
+
+Epic covering the progress of the playground:
+
+- https://github.com/vectordotdev/vector/issues/14652
+
+List of relevant stretch goals:
+
+- Add syntax highlighting
+- Add keyboard shortcut support
+- Add advanced IDE features (auto-completion, inline docs, etc...)
+- Add button to convert tested program to working Vector config.toml file
+- Support multiple VRL (Vector) versions
+
+## Cross cutting concerns
+
+- https://github.com/vectordotdev/vector/issues/14714.
+
+## Scope
+
+### In scope
+
+- Leveraging a web framework that would facilitate a maintainable playground.
+- Styling the playground to abide by Vector style guides.
+- Adding UI components that support stretch goals such as Dialogs, ToolBars, MenuItems, etc.
+- Adding CI steps to release `/lib/vrl/web-playground/public` for running VRL under different versions of Vector.
+- Migrating current MVP features to new UI.
+- Setting up the long-term repo for the web playground.
+
+### Out of scope
+
+- Creating a language server for VRL.
+
+## Pain
+
+The alternatives are covered in a section below but the main
+concerns have to do with.
+
+- Maintainability. We must ensure other engineers can build on top of the playground in the long term.
+
+- Timeline. There's six weeks left in Jonathan's internship
+
+- Lack of customizeability, current alternatives will provide bottlenecks in the long term future for adding more features
+
+## Proposal
+
+### User Experience
+
+- The expansion of the MVP UI is necessary to create more robust features that tightly integrates with VRL and Vector so users can get a clear
+
+### Implementation
+
+The new UI will set up the building blocks for a robust
+playground experience. Live deploy, auto completion, in line docs,
+and more desired features will be quicker to implement in a
```

Comment:
It looks like this section isn't complete?

I was expecting some details on _how_ this will be accomplished. For example, will it be using `React`? And if so, are we writing most component ourselves? Are there any we will use from somewhere else?